packages/ethereum/contracts/src/utils/EnumerableTargetSet.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.8.0) (utils/structs/EnumerableSet.sol)
+// This file was procedurally generated from scripts/generate/templates/EnumerableSet.js.
+
+pragma solidity ^0.8.0;
+
+import "./TargetUtils.sol";
+
+struct TargetSet {
+    // Storage of set values
+    Target[] _values;
+    // Position of the value in the `values` array, plus 1 because index 0
+    // means a value is not in the set.
+    // the key is not `Target` but the first 160 bits converted to address
+    mapping(address => uint256) _indexes;
+}
+
+/**
+ * @dev Library for managing
+ * https://en.wikipedia.org/wiki/Set_(abstract_data_type)[sets] of primitive
+ * types.
+ *
+ * Adapted from OpenZeppelin's EnumerableSet and EnumerableMap (`AddressToUintMap`)
+ * library, for TargetDefaultPermissions type.
+ */
+library EnumerableTargetSet {
+    using TargetUtils for Target;
```

Comment:
Do you really need the `values()` function? This requires two slots for storing the same data, meaning 40k gas per permission.

Otherwise we could easily increase / decrease a counter whenever we insert / remove a value

Reply:
The idea is to store the mapping in an array so that the current version of UI can query the information with one single RPC call.